metadata/lib.rs

```diff
@@ -92,27 +93,84 @@ impl ContractMetadata {
             abi,
         }
     }
+
+    pub fn remove_source_wasm_attribute(&mut self) {
+        self.source.wasm = None;
+    }
+}
+
+/// Representation of the Wasm code hash.
+#[derive(Debug, Eq, PartialEq)]
+pub struct CodeHash(pub [u8; 32]);
+
+impl Serialize for CodeHash {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serialize_as_byte_str(&self.0[..], serializer)
+    }
 }
 
 #[derive(Debug, Serialize)]
 pub struct Source {
-    #[serde(serialize_with = "serialize_as_byte_str")]
-    hash: [u8; 32],
+    #[serde(skip_serializing_if = "Option::is_none")]
+    hash: Option<CodeHash>,
```

Comment:
When is this ever `None`? I guess it is always useful even if we provide the sources.

Reply:
Discussion happened above, [here](https://github.com/paritytech/cargo-contract/pull/97#discussion_r519683048).

Reply:
What I read from the discussion is that we still always want the `hash` field to be available. And also I don't think that we need `--metadata-only`. Please convince me of the opposite by coming up with a useful use case.

Reply:
>What I read from the discussion is that we still always want the hash field to be available.

I understood the discussion in the way that we want to omit the hash:
>It is not required, currently. It's just there as a nice to have. In the future the client tools could use that (if present) to verify the metadata matches the deployed code.
>
>_So I would say that if metadata-only is specified then that field should be omitted_, even if the wasm is already there because it is not guaranteed to match.